SteamRPC.js

```diff
@@ -110,25 +107,64 @@ async function pollSteamPresence(steamUserId, profiles) {
         gameStatus = "";
         let resJson = await res.json();
         if (resJson.in_game) {
-            gameStatus = `${resJson.in_game.name} (not supported)`;
-            if (resJson.in_game?.logo && resJson.in_game?.rich_presence) {
-                let curr_appid = resJson.in_game.logo.split("/apps/")[1].split("/")[0];
-                let curr_rpc = resJson.in_game.rich_presence;
-                debugLine = "Current Steam RPC Status: " + resJson.in_game.rich_presence;
-                if (profiles.hasOwnProperty(curr_appid)) {
-                    let profile = profiles[curr_appid];
-                    try {
-                        gameStatus = profile.title;
-                        let translatedDiscordRPC = profile.translateSteamPresence(curr_rpc);
-                        if (typeof translatedDiscordRPC !== "object") throw `Profile returned '${typeof translatedDiscordRPC}' instead of an object.`;
-                        discordRPCClient.user?.setActivity(translatedDiscordRPC);
-                    }
-                    catch (codeErr) {
-                        throw new Error(`A code error has occured in the game profile for '${profile.title}'!`, {cause: codeErr});
+            gameStatus = resJson.in_game.name;
+
+            if(resJson.in_game?.rich_presence === "")
+                debugLine = "There is no RPC status for this game.";
+            else {
+                if (resJson.in_game?.logo && resJson.in_game?.rich_presence) {
+                    let currentAppId = resJson.in_game.logo.split("/apps/")[1].split("/")[0];
+                    let currentSteamStatus = resJson.in_game.rich_presence;
+                    debugLine = "Current Steam RPC Status: " + resJson.in_game.rich_presence;
+
+                    if (profiles.hasOwnProperty(currentAppId)) {
+                        let profile = profiles[currentAppId];
+                        try {
+                            gameStatus = profile.title;
+                            let clientId;
+
+                            if(profile.clientId !== undefined)
+                                clientId = profile.clientId;
+                            else
+                                clientId = config.default_client_id;
+    
+                            if(discordRPCClient === null) 
+                                initRPC(clientId);
+                            else if(discordRPCClient.clientId !== clientId) {
+                                discordRPCClient.login({clientId});
```

Comment:
The login method doesn't accept a clientId but some other (irrelevant) options. I think the entire discordRPCClient has to be destroyed and reinitialized each time we want to switch the clientId, or we make more then one discordRPCClient object (afaik the newest rich presence app will take priority, but I think having a single instance will be better).